src/main/java/io/supertokens/storage/postgresql/ConnectionPool.java

```
@@ -198,6 +198,16 @@ public static Connection getConnection(Start start) throws SQLException {
         return getInstance(start).hikariDataSource.getConnection();
     }
 
+    public static HikariDataSource getHikariDataSource(Start start) throws SQLException {
+        if (getInstance(start) == null || getInstance(start).hikariDataSource == null ) {
+            throw new IllegalStateException("Please call initPool before getHikariDataSource");
+        }
+        if (!start.enabled) {
+            throw new SQLException("Storage layer disabled");
+        }
+        return getInstance(start).hikariDataSource;
```

do we expect this to always be initialised when this function is called? If so, and if it's not initialilsed, we should throw a runtime exception in this function.